_source/_posts/2022-01-19-react-storybook.md

```diff
@@ -0,0 +1,688 @@
+
+# Adding authentication with Okta to the application
+
+You have created a page that uses an `useAuth` hook to manage user authentication. For the Storybook stories, you made a mock implementation of this hook. This section will show you how to implement the hook using Okta's authentication service. First, register the application with Okta.
+
+{% include setup/cli.md type="spa" framework="React" loginRedirectUri="http://localhost:3000/callback" logoutRedirectUri="http://localhost:3000" %}
+
+Next, install the necessary libraries. Open the terminal and run the command below.
+
+```bash
+npm install -E @okta/okta-react@6.4.1 @okta/okta-auth-js@5.10.0 react-dom@17.0.2 react-router-dom@5.3.0
+```
+
+Open the file `src/index.js` and modify its contents to match the code below.
+
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import { App } from './App';
+import { Page } from './stories/Page';
+import reportWebVitals from './reportWebVitals';
+import { BrowserRouter as Router, Route, useHistory } from 'react-router-dom';
+import { LoginCallback, SecureRoute, Security } from '@okta/okta-react';
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
+import { useAuth } from './auth';
+
+const oktaAuth = new OktaAuth({
+  issuer: 'https://{yourOktaDomain}/oauth2/default',
+  clientId: '{clientID}',
+  redirectUri: `/callback`,
+});
+
+function SecuredRoutes(props) {
+  const history = useHistory();
+  const restoreOriginalUri = async (_oktaAuth, originalUri) => {
+    history.replace(toRelativeUrl(originalUri || '/', window.location.origin));
+  };
+  
+  return (
+    <Security oktaAuth={oktaAuth} restoreOriginalUri={restoreOriginalUri}>
+      <Route path="/" exact render={(props) => <App {...props} useAuth={useAuth}/>} />
+      <SecureRoute path="/converter" exact render={(props) => <Page {...props} useAuth={useAuth}/>} />
+      <Route path="/callback" component={LoginCallback} />
+    </Security>
+  );
+}
+
+ReactDOM.render(
+  <React.StrictMode>
+    <Router>
+      <SecuredRoutes />
+    </Router>
+  </React.StrictMode>,
+  document.getElementById('root')
+);
+
+reportWebVitals();
+```
+
+Here `{yourClientID}` is the client ID that you obtained earlier and `{yourOktaDomain}` is your Okta domain. This change does several things. The `oktaAuth` instance provides a global authentication singleton. The main `render` function now contains a `Router` element that allows the application to navigate different routes. Finally, `SecuredRoutes` is a component that wraps the routes in a `Security` component. This component makes a `useOktaAuth` hook available for all components contained within it. Inside this component, you define the routes. Note how you pass in a `useAuth` hook into the `App` and the `Page` components. Create a new file `src/auth.js` and add the following code to implement this hook.
```

Comment:
```suggestion
Here `{yourClientID}` is the client ID that you obtained earlier and `{yourOktaDomain}` is your Okta domain. This change does several things. The `oktaAuth` instance provides a global authentication singleton. The main `render` function now contains a `Router` element that allows the application to navigate different routes. Finally, `SecuredRoutes` is a component that wraps the routes in a `Security` component. This component makes a `useOktaAuth` hook available for all components contained within it. Inside this component, you define the routes. Note how you pass a `useAuth` hook into the `App` and the `Page` components. Create a new file `src/auth.js` and add the following code to implement this hook.
```
removes the double "in" - since you have it with "into"